### DISCUSSION TITLE
Delayed constraints with polling

### DISCUSSION THREAD

**Discussion by Riddleware** [xunit/xunit#2363] [Question] [answered]
Hi, hopefully this is the right place for this, apologies if it isn't. 
Started work for a new project which uses xUnit for unit testing. I find there are a few features I really miss from using NUnit (which I'm more accustomed to). The main one is delayed constraints, such as 

`Assert.That().Constraint().After(3).Seconds.PollEvery(250).MilliSeconds);`

or even 

`Assert.That().Constraint().After(3, 250);`

This is really useful when testing some functionality when we have no way to tell execution has finished (generally for integration or component testing - I could look at log files / entries but find that to be inelegant as the log is not 'real' functionality that we should be testing against - unless specifically testing the logs are written correctly, and there might not be a suitable log entry, but I digress).

Is there any plan to implement this, or is its omission by design (ie. is there a reason xUnit devs do not like this type of thing?), as I was surprised by its absence, which means I have to write my own code for various scenarios, generally a loop / sleep type thing, in order to achieve the same thing. Having this in xUnit would save whole minutes of my time!!

Thanks & Best Regards.

**Comment by bradwilson** [comment, marked as answer]
Sorry for the delayed response.

I'm not averse to having something like this in the base framework. I don't know what exact shape the API might take, since we don't have fluent APIs. Please open an issue and I'll put it on the backlog. Thanks!
